src/Microsoft.TestPlatform.CrossPlatEngine/Execution/BaseRunTests.cs

```diff
@@ -504,6 +511,11 @@ private void SetAdapterLoggingSettings()
                     attachments,
                     elapsedTime);
 
+                if (lastChunk.Count() > 0 && this.packages != null)
+                {
+                    lastChunk.ToList().ForEach(tr => tr.TestCase.Source = this.packages.FirstOrDefault());
```

Comment:
Multiple enumeration in Count and ToList. Fix string comparison

Reply:
Changed Count to Any(), 
We are not comparing bu assigning.